pkg/querier/queryrange/codec.go

```diff
@@ -1515,10 +1563,14 @@ type paramsLabelWrapper struct {
 	*LabelRequest
 }
 
-func (p paramsLabelWrapper) Query() string {
+func (p paramsLabelWrapper) QueryString() string {
 	return p.GetQuery()
 }
 
+func (p paramsLabelWrapper) GetExpression() syntax.Expr {
```

Comment:
Hhmm, does it make sense for these to even implement the interface?
Seems like we need a parent interface here, and `GetExpression()`, `QueryString()` etc can move into a sub-interface of that

Reply:
Ok, I don't this is feasible in this change. [stats](https://github.com/grafana/loki/blob/main/pkg/querier/queryrange/stats.go#L54) is using the params and they must have an expression for the [RecordRangeAndInstantQueryMetrics](https://github.com/grafana/loki/blob/main/pkg/logql/metrics.go#L87). I can do a follow up and change them so they use their own interface.

Reply:
Sounds good, it's a non-blocking nit

Reply:
Yeah lets clean these up in a follow up. Nice that we have a better sense of how all this should fit together as a result of this PR :+1: